### DISCUSSION TITLE
Useful bookmarklet to check a Github repository against SPI

### DISCUSSION THREAD

**Discussion by daveverwer** [SwiftPackageIndex/SwiftPackageIndex-Server#1641] [Show and tell]
Am I posting this here so that it's saved somewhere "safe"? Maybe 😅
Might it be useful for other people anyway? Also, maybe 😅

```javascript
var loc = location.href
const ghRegex = /^http[s]?:\/\/.*github.com\//i
const spiRegex = /^http[s]?:\/\/.*swiftpackageindex.com\//i
if (loc.match(ghRegex)) {
  location.href = loc.replace(ghRegex, 'https://swiftpackageindex.com/')
} else if (loc.match(spiRegex)) {
  location.href = loc.replace(spiRegex, 'https://github.com/')
}
```

... and as a bookmarklet ...

```javascript
javascript:(function()%7Bvar%20loc%20%3D%20location.href%0Aconst%20ghRegex%20%3D%20%2F%5Ehttp%5Bs%5D%3F%3A%5C%2F%5C%2F.*github.com%5C%2F%2Fi%0Aconst%20spiRegex%20%3D%20%2F%5Ehttp%5Bs%5D%3F%3A%5C%2F%5C%2F.*swiftpackageindex.com%5C%2F%2Fi%0Aif%20(loc.match(ghRegex))%20%7B%0A%20%20location.href%20%3D%20loc.replace(ghRegex%2C%20'https%3A%2F%2Fswiftpackageindex.com%2F')%0A%7D%20else%20if%20(loc.match(spiRegex))%20%7B%0A%20%20location.href%20%3D%20loc.replace(spiRegex%2C%20'https%3A%2F%2Fgithub.com%2F')%0A%7D%7D)()%3B
```

It doesn't support staging right now but it could quite easily. If anyone makes enhancements and wants to make a new bookmarklet, [this tool was very useful](https://caiorss.github.io/bookmarklet-maker/).

**Comment by Sherlouk** [comment]
I didn't even know this was a thing you could do... woah 👀

**Comment by finestructure** [comment]
This is great. Do we even need need staging? The use case I had in mind was to replace GitHub for a quick check if we have a package and add it if we don't. While staging and prod should be the same pretty much always, prod _would_ be the canonical source.

**Comment by finestructure** [comment]
I’ve turned this into a Shortcut so it’s easier to invoke from iOS (I couldn’t figure out how to invoke the bookmarklet from an action)

https://www.icloud.com/shortcuts/c95303c3ef704525aaebd70347e4e693

![FAEE5AA4-7921-4B22-B91F-491AAB494547](https://user-images.githubusercontent.com/65520/160283070-7e06b379-8e20-48dc-a52e-592e333716a1.jpeg)
